### PR TITLE
Minimize shade bouncycastle dependencies

### PR DESCRIPTION
**Pull request by moarychan**
# Description

Cherry-pick from https://github.com/Azure/azure-sdk-for-java/pull/42307, you can find some discussions in the origin PR, this PR is targeted to main branch.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by moarychan** [comment]
Hi @srnagar and @g2vinay , 

We have communicated about shade plugin configuration, this PR is to optimize the configuration to stop the deps transitivity, it's targeted to main branch, please help review it! Many thanks!

File: sdk/keyvault/azure-security-keyvault-jca/pom.xml [hunk 1]
```diff
@@ -135,13 +136,20 @@
...
           <execution>
+            <id>minimize-shade</id>
             <goals>
               <goal>shade</goal>
             </goals>
             <configuration>
               <shadeTestJar>false</shadeTestJar>
               <createSourcesJar>${createSourcesJar}</createSourcesJar>
               <shadeSourcesContent>true</shadeSourcesContent>
+              <minimizeJar>true</minimizeJar>
```
Comment (rujche) [inline sdk/keyvault/azure-security-keyvault-jca/pom.xml:147]: Why other dependencies can not apply `minimizeJar`?

Comment (saragluna) [reply sdk/keyvault/azure-security-keyvault-jca/pom.xml:147]: I think we do could do the same for others, but to minimize the changes in this PR, we probably could do it later.

**Review by saragluna — APPROVED** [review]

**Comment by moarychan** [comment]
Hi @srnagar and @g2vinay , thanks for your support! 
I'm going to release the new version 2.10.0 of JCA and preparing to merge the PR. Thanks!

**Review by rujche — APPROVED** [review]
LGTM.

File: NOTICE.txt [hunk 2]
```diff
@@ -462,6 +462,23 @@ rights to this code.
 You should have received a copy of the CC0 legalcode along with this
 work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 
+License notice for Bouncy Castle
+------------------------------------------------------------------------------
+
+Copyright (c) 2000 - 2023 The Legion of the Bouncy Castle Inc. (https://www.bouncycastle.org)
```
Comment (g2vinay) [inline NOTICE.txt:468]: This should be 2000 - 2024 I think

**Review by srnagar — APPROVED** [review]
LGTM!

**Review by g2vinay — APPROVED** [review]

**Pull request merged**